app-backend/api/src/main/scala/datasource/Routes.scala

```diff
@@ -52,30 +52,34 @@ trait DatasourceRoutes extends Authentication
     get {
       rejectEmptyResponse {
         complete {
-          readOne[Datasource](Datasources.getDatasource(datasourceId))
+          readOne[Datasource](Datasources.getDatasource(datasourceId, user))
         }
       }
     }
   }
 
   def createDatasource: Route = authenticate { user =>
     entity(as[Datasource.Create]) { newDatasource =>
-      onSuccess(write[Datasource](Datasources.insertDatasource(newDatasource, user))) { datasource =>
-        complete(datasource)
+      authorize(user.isInRootOrSameOrganizationAs(newDatasource)) {
```

Comment:
Am I right in interpreting that:

- `authenticate` == is the user logged in / are API tokens legit?
- `authorize` == does the user have permission to access this resource?